bpf/lib/maps.h

```diff
@@ -121,18 +121,22 @@ struct bpf_elf_map __section_maps tunnel_endpoint_map = {
 
 struct bpf_lpm_trie_key6 {
 	struct bpf_lpm_trie_key lpm_key;
+	__u16 pad1;
+	__u16 pad2;
 	union v6addr lpm_addr;
 };
 
 struct bpf_lpm_trie_key4 {
 	struct bpf_lpm_trie_key lpm_key;
+	__u16 pad1;
+	__u16 pad2;
 	__be32 lpm_addr;
 };
 
 static __always_inline int lpm6_map_lookup(struct bpf_elf_map *map,
 					   union v6addr *addr, __u32 prefix)
 {
-	struct bpf_lpm_trie_key6 key = { { prefix }, *addr };
+	struct bpf_lpm_trie_key6 key = { { prefix + sizeof(__u32) }, 0, 0, *addr };
```

Comment:
Shouldn't this be: sizeof(__u32) * 8 ? Call-sites use prefix value of either 32 or 128, and if we extend it by 4 bytes, it should say either 64 or 160.

Reply:
right, I just thought about this last night and apparently didn't think clearly enough. I'll fix it.